src/prefect/engine.py

```diff
@@ -343,24 +343,22 @@ async def begin_flow_run(
             report_flow_run_crashes(flow_run=flow_run, client=client)
         )
 
-        # Create a task group for background tasks
-        flow_run_context.background_tasks = await stack.enter_async_context(
-            anyio.create_task_group()
+        flow_run_context.task_runner = await stack.enter_async_context(
+            flow.task_runner.start()
         )
 
         # If the flow is async, we need to provide a portal so sync tasks can run
-        flow_run_context.sync_portal = (
-            stack.enter_context(start_blocking_portal()) if flow.isasync else None
+        flow_run_context.sync_portal = stack.enter_context(start_blocking_portal())
+
+        # Create a task group for background tasks
+        flow_run_context.background_tasks = await stack.enter_async_context(
+            anyio.create_task_group()
```

Comment:
Moved this after `task_runner` because if not, it complains that the task runner hasn't started yet

Reply:
Huh? Background tasks shouldn't require the task runner to be started.

Reply:
With the current changes:
```
File ~/Applications/python/prefect/src/prefect/task_runners.py:250, in ConcurrentTaskRunner.submit(self, key, call)
    249 if not self._started:
--> 250     raise RuntimeError(
    251         "The task runner must be started before submitting work."
    252     )
```

And I'm using background_tasks here because:
```
RuntimeError: Task <Task pending name='' coro=<create_task_run_then_submit() running at /Users/mz/dev/prefect/src/prefect/engine.py:940> cb=[TaskGroup._spawn.<locals>.task_done() at /opt/homebrew/Caskroom/miniconda/base/envs/orion-dev-38/lib/python3.8/site-packages/anyio/_backends/_asyncio.py:726]> got Future <Future pending> attached to a different loop
```

Reply:
Hm interesting. Perhaps this is because of context exit, i.e. the task runner exits before the background tasks are all done if its context is entered second.